src/raster/processing/dem/options.rs

```diff
@@ -0,0 +1,172 @@
+use std::fmt::{Display, Formatter};
+use std::marker::PhantomData;
+use std::ptr;
+use std::ptr::NonNull;
+
+use gdal_sys::{
+    GDALDEMProcessingOptions, GDALDEMProcessingOptionsFree, GDALDEMProcessingOptionsNew,
+};
+
+use crate::cpl::CslStringList;
+use crate::errors;
+use crate::utils::_last_null_pointer_err;
+
+/// Payload for [`GDALDEMProcessing`]. Intended for internal use only.
+pub struct GdalDEMProcessingOptions<'opts>(
+    NonNull<GDALDEMProcessingOptions>,
+    PhantomData<&'opts CslStringList>,
+);
+
+impl<'opts> GdalDEMProcessingOptions<'opts> {
+    pub fn new(opts: &'opts CslStringList) -> errors::Result<Self> {
+        let popts = unsafe { GDALDEMProcessingOptionsNew(opts.as_ptr(), ptr::null_mut()) };
+        if popts.is_null() {
+            return Err(_last_null_pointer_err("GDALDEMProcessingOptionsNew"));
+        }
+        Ok(Self(unsafe { NonNull::new_unchecked(popts) }, PhantomData))
+    }
+
+    pub fn as_ptr(&self) -> *const GDALDEMProcessingOptions {
+        self.0.as_ptr()
+    }
+}
+
+impl Drop for GdalDEMProcessingOptions<'_> {
+    fn drop(&mut self) {
+        unsafe { GDALDEMProcessingOptionsFree(self.0.as_ptr()) };
+    }
+}
+
+/// DEM processor mode, to stringify and pass to [`gdal_sys::GDALDEMProcessing`].
+#[derive(Debug, Clone, Copy)]
+pub enum DemAlg {
+    Aspect,
+    ColorRelief,
+    Hillshade,
+    Roughness,
+    Slope,
+    Tpi,
+    Tri,
+}
+
+impl Display for DemAlg {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::ColorRelief => f.write_str("color-relief"),
+            _ => {
+                let s = format!("{self:?}").to_lowercase();
```

Comment:
Man page says `TPI` and `TRI`, if this works, then I suspect GDAL isn't that case-sensitive here.

Reply:
https://github.com/metasim/gdal/pull/1